### PR TITLE
DAOS-9548 placement: add container property DAOS_PROP_CO_PERF_DOMAIN

### PR DESCRIPTION
**Pull request by liuxuezhao**
Can create container with "--properties=perf_domain:[root/group]", if
    not specified will inherit from pool property.
    Restrict valid value of perf_domain to be root or group.
    Add daos_obj_md::omd_pdom_lvl to pass perf_domain lvl for layout
    generation.
    Change DAOS_PROP_PO_RP_PDA_DEFAULT to (-1).
    Some other detailed refining.
    Add jump_map_pda_layout.c to show layout.
    
    Required-githooks: true
    Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by mjmac — COMMENTED** [review]
The Go changes LGTM.

File: src/container/srv_container.c [hunk 1]
```diff
@@ -737,6 +738,14 @@ cont_create_prop_prepare(struct ds_pool_hdl *pool_hdl,
 		entry_def->dpe_val = pool_hdl->sph_pool->sp_rp_pda;
 	}
 
+	entry_def = daos_prop_entry_get(prop_def, DAOS_PROP_CO_PERF_DOMAIN);
+	if (pool_hdl->sph_global_ver > 0)
```
Comment (wangshilong) [inline src/container/srv_container.c:742]: I think PDA is not in release 2.4, so we might need bump DAOS_POOL_GLOBAL_VERSION to 3, and then here it should.

if (pool_hdl->sph_global_ver > 2)
        D_ASSERT(...)

Comment (liuxuezhao) [reply src/container/srv_container.c]: OK, refreshed as the suggestion. Thanks!

**Review by wangshilong — APPROVED** [review]

**Comment by liuxuezhao** [comment]
only one test failure -
ASSERT: result != -DER_TX_RESTART
src/object/obj_tx.c:3633: error: Failure!Test teardown failed

due to https://daosio.atlassian.net/browse/DAOS-13914

**Review by wangdi1 — DISMISSED** [review]

File: src/placement/jump_map.c [hunk 2]
```diff
@@ -79,10 +84,15 @@ jm_obj_pd_init(struct pl_jump_map *jmap, struct daos_obj_md *md, struct pool_dom
...
 	D_ASSERTF(doms_per_pd >= 1, "bad dom_nr %d, pd_nr %d\n", dom_nr, pd_nr);
 
-	if (jmop->jmop_grp_size == 1) /* non-replica */
+	if (jmop->jmop_grp_size == 1) { /* non-replica */
 		pd_grp_size = min(md->omd_pda, pds->do_target_nr);
-	else
-		pd_grp_size = min(md->omd_pda, doms_per_pd);
+	} else {
+		if (md->omd_pda == DAOS_PROP_PDA_MAX)
+			pd_grp_size = jmop->jmop_grp_size;
```
[the remark this answers is not in the document]
Comment (liuxuezhao) [reply src/placement/jump_map.c:91]: for group_size > 1 case (replica or EC), if the PDA is MAX (-1), it means put all shards in the RDG to one PD, but cannot exceed doms_per_pd.

**Review by wangdi1 — DISMISSED** [review]

**Review by wangdi1 — APPROVED** [review]

**Comment by liuxuezhao** [comment]
just a rebase to fix conflict.

**Review by wangdi1 — APPROVED** [review]

**Comment by liuxuezhao** [comment]
@daos-stack/daos-gatekeeper this PR have pending quite long time, could you please land it?

**Pull request merged**